### PR TITLE
Implement changes to Clubs form

### LINKED ISSUE #910: Changes to Clubs form
Placeholder ticket, pending @thornet and @amirad's research into what information is necessary to know whether a Club meets our criteria.

### PR DESCRIPTION
**Pull request by toolness**
This fixes #910, here's a screenshot:

### CODE REVIEW

File: pages/clubs.jsx [hunk 1]
```diff
@@ -323,6 +324,11 @@ var ModalAddOrChangeYourClub = React.createClass({
       'name', 'website', 'description', 'location', 'latitude', 'longitude'
     ));
     var validationErrors = validateClub(clubState);
+
+    if (!this.props.club && !this.state.hasReadFactSheet) {
```
Comment (toolness) [inline pages/clubs.jsx]: The `!this.props.club` is checking to see if we're the "add your club" form or the "change your club" form--we don't show the checkbox in the latter.

**Comment by toolness** [comment]
Ok @mmmavis, I think this is ready for review! You can see the [HTML Form Validation section of `CONTRIBUTING.md`](https://github.com/mozilla/teach.webmaker.org/blob/develop/CONTRIBUTING.md#html5-form-validation) for info on how to easily make sure our forms work, too.

File: test/browser/clubs-page.test.jsx [hunk 2]
```diff
@@ -298,6 +298,15 @@ describe("ClubsPage.ModalAddOrChangeYourClub", function() {
       should(modal.state.latitude).equal(null);
     });
 
+    it("requires user to read fact sheet", function() {
+      teachAPI.emit('username:change', 'foo');
```
Comment (mmmavis) [inline test/browser/clubs-page.test.jsx]: I see `username:change` being emitted across several places here - sorry never paid too much attention on the Teach API end before.  Does `teachAPI.emit('username:change', 'foo')` mean we are now logged in as user "foo"?

Comment (toolness) [inline test/browser/clubs-page.test.jsx]: yup exactly!

**Comment by mmmavis** [comment]
coooooooooool R+ thanks @toolness 
(yes, one more PR reviewed :grin: )

**Pull request merged**